gridcomps/Cap/MAPL_Cap.F90

```diff
@@ -288,16 +289,40 @@ subroutine run_model(this, comm, unusable, rc)
       call MPI_COMM_RANK(comm, rank, status)
       _VERIFY(status)
 
+      ! We look to see if the user has set an environment variable for the
+      ! name of the ESMF configuration file. If they have, we use that. If not,
+      ! we use the default of "ESMF.rc" for backward compatibility
+
+      ! Step one: default to ESMF.rc
+
+      esmfConfigFile = 'ESMF.rc'
+      esmfConfigFileLen = len(esmfConfigFile)
+
+      ! Step two: get the length of the environment variable
+      call get_environment_variable('ESMF_CONFIG_FILE', length=esmfConfigFileLen, status=status)
+      ! Step three: if the environment variable exists, get the value of the environment variable
+      if (status == 0) then
+         ! We need to deallocate so we can reallocate
+         deallocate(esmfConfigFile)
+         allocate(character(len = esmfConfigFileLen) :: esmfConfigFile)
+         call get_environment_variable('ESMF_CONFIG_FILE', value=esmfConfigFile, status=status)
+      end if
+
       if (rank == 0) then
-         inquire(file='ESMF.rc', exist=file_exists)
+         inquire(file=esmfConfigFile, exist=esmfConfigFileExists)
       end if
-      call MPI_BCAST(file_exists, 1, MPI_LOGICAL, 0, comm, status)
+      call MPI_BCAST(esmfConfigFileExists, 1, MPI_LOGICAL, 0, comm, status)
       _VERIFY(status)
+      call MPI_BCAST(esmfConfigFile, esmfConfigFileLen, MPI_CHARACTER, 0, comm, status)
+      _VERIFY(status)
+
+      lgr => logging%get_logger('MAPL')
 
       ! If the file exists, we pass it into ESMF_Initialize, else, we
       ! use the one from the command line arguments
-      if (file_exists) then
-         call ESMF_Initialize (configFileName='ESMF.rc', mpiCommunicator=comm, vm=vm, _RC)
+      if (esmfConfigFileExists) then
+         call lgr%info("Using ESMF configuration file: %a", esmfConfigFile)
+         call ESMF_Initialize (configFileName=esmfConfigFile, mpiCommunicator=comm, vm=vm, _RC)
```

Comment:
If `esmfConfigFile` is allocatable, then you do not need a separate call to ESMF_Initialized.  An unallocated actual appears as a non-present dummy.  So this conditional can be simplified.   (You may still want the conditional for the lgr bit though.)